src/pages/Teacher.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import { Chart } from "react-google-charts";
+import useLoadChartData from "../hooks/useLoadData";
+import useLoadCourseSearch from "../hooks/useLoadCourseSearch";
+import useLoadStudent from "../hooks/useLoadStudents";
+import axios from "axios";
+import { AuthContext } from "../pages/AuthProvider";
+import { useContext } from "react";
+
+const Teacher = () => {
+  const { currentUser } = useContext(AuthContext);
+  const { initialGid, initialCourse } = useLoadChartData(0);
+  const [count, setCount] = useState(0);
+  const [gid, setGid] = useState(null);
+  const { courseOptions, courseGidMap } = useLoadCourseSearch();
+  const { studentCount } = useLoadStudent();
+  const [courseName, setCourseName] = useState("");
+  const [remainingCount, setRemainingCount] = useState(0);
+  const [data, setData] = useState([]);
+  const spreadSheetId = "16HTIiiOq82Tm1tLHRQcr_8YJnO81QxZOOBOfR4hU3zc";
```

Comment:
I can see the same ID in 3 different files here. This is repetition and we should abstract it somewhere to only be defined once. `.env` would probably be the best bet for now if it is static